Rmd/tgx_ddi.Rmd

```diff
@@ -0,0 +1,555 @@
+df$ID <- sub("_.*", "", df$ID) # remove everything after the underscore in the first column of df
+
+# Convert all columns except ID to numeric, collapse probes to genes
+# Probes are already collapsed to genes in write_additional_output.R lines 30-32
+# So taking means here probably is not necessary
+df <- df %>%
+  mutate(across(-ID, as.numeric)) %>%
+  group_by(ID) %>%
+  summarise_all(mean)
+
+## Average the dose replicates
+# Replace sample names with info from the metadata column used for deseq2 design
+designcol <- params$design
+df_no_id <- select(df, -ID)
+name_mapping <- setNames(exp_metadata[[designcol]], exp_metadata$original_names)
+colnames(df_no_id) <- name_mapping[colnames(df_no_id)]
+
+# Calculate the row-wise mean for each unique base name, add ID column back on
+df_means <- df_no_id %>%
+  split.default(names(df_no_id)) %>%
+  map_df(rowMeans, na.rm = TRUE)
+df_means <- bind_cols(df["ID"], df_means)
+
+# Subset dataframe to just the reports group in case2
+if (is.na(params$deseq_facet) && !is.na(params$reports_facet)) {
+  # Just want columns for the report facet plus the control
+  filter_columns <- contrasts_subset[[1]]
+  filter_columns <- c("ID", filter_columns, control)
+  #Replace "." with "-" in filter_columns
+  filter_columns <- gsub("\\.", "-", filter_columns)
+
+
+  # Subset the columns from the formatted df
+  df_means <- df_means %>%
+    select(all_of(filter_columns))
+}
+
+# Log2 ratios vs control (from contrasts)
+log2ratios <- df_means
+log2ratios[, -1] <- sweep(log2ratios[, -1], 1, log2ratios[[control]], `-`)
+# Remove the control column
+log2ratios <- select(log2ratios, -one_of(control))
+
+log10ratios <- log2ratios %>%
+  mutate_at(vars(-ID), ~ log10(2^.))
+```
+
+```{r classifier}
+
+classifier_path <- file.path(params$projectdir, "resources", "tgx-ddi_biomarker-classifier_Li-2015.txt")
+
+classifier <- read.delim(classifier_path, sep = "\t")
+
+colnames(classifier)[11:39] <- c("5 FU", "AraC", "Bleomycin", "Camptothecin",
+                                 "Cisplatin", "Etoposide", "Hydrogen Peroxide", "Hydroxyurea",
+                                 "Methotrexate",  "MMS", "Potassium Chromate VI", "Sodium Arsenite",
+                                 "X ray", "2 DG", "Apicidin","Antimycin A", "Cadmium Chloride",
+                                 "Colchicine", "Docetaxel", "EtOH 2%", "EtOH 4%",
+                                 "HC toxin", "Heat Shock", "Oxamflatin", "Paclitaxel",
+                                 "Thapsigargin", "TSA", "Tunicamysin", "Vinblastine")
+
+# Fix annotations depending on temposeq kit
+# Need to talk to Andrew to find additional fixes
+if (params$biospyder_manifest_file == "Human_Whole_Transcriptome_2.0_standardized.csv") {
+  classifier$ID[8] <- "TIGAR"
+  classifier$ID[22] <- "AMER1"
+}
+
+DDI_chems <- c("5 FU",
+  "AraC",
+  "Bleomycin",
+  "Camptothecin",
+  "Cisplatin",
+  "Etoposide",
+  "Hydrogen Peroxide",
+  "Hydroxyurea",
+  "Methotrexate",
+  "MMS",
```

Comment:
Some of these get abbreviations while some get full names - maybe update to be consistent (I know this might have to be harmonized with the column names in the source data too)

Reply:
Yeah, the new names were taken straight from Andrew's code. I'll need to dig to find full chemical names (in the original paper, I guess?) but it should be possible.

I have concerns about renaming the genes to match temposeq annotations. It relies on the manifest files being named exactly as expected. I guess the standardized manifest files are on our EHSRB repo, and it would be weird for someone to download the csv and then rename them, but not unheard of. Renaming genes is definitely necessary, though, because I've just realized that the Genetox21 stuff I ran this morning is only using 53 of the 54 biomarker genes.

Reply:
Could we have a table of aliases? Some kind of lookup values, such that it would check for the right names?